### PR TITLE
Create final checkpoint with max updates, samples or epochs.

### PR DESCRIPTION
**Pull request by tdomhan**
Previously there was an edge case with very small training sets where if the `--max-updates` was below the `--checkpoint-interval` no checkpoint was written. With this change we make sure to write a final checkpoint so that all gradient updates performed are actually capture by checkpoints.

#### Pull Request Checklist ##
- [x] Changes are complete (if posting work-in-progress code, prefix your pull request title with '[WIP]'
until you can check this box.
- [x] Unit tests pass (`pytest`)
- [x] Passed code style checking (`./style-check.sh`)
- [x] Updated major/minor version in `sockeye/__init__.py`. Major version bump if this is a backwards incompatible change.
- [x] Updated CHANGELOG.md

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by fhieber — APPROVED** [review]
LGTM

**Comment by fhieber** [comment]
@tdomhan Actually, can we create a small integration test to test this behavior?

File: test/integration/test_seq_copy_int.py [hunk 1]
```diff
@@ -51,7 +51,9 @@
      " --weight-tying-type src_trg_softmax"
      " --weight-init-scale=3.0 --weight-init-xavier-factor-type=avg"
      " --batch-size 2 --max-updates 2 --batch-type sentence --decode-and-evaluate 0"
-     " --checkpoint-interval 2 --optimizer adam --initial-learning-rate 0.01",
+     # Note: We se the checkpoint interval > max updates in order to make sure we create a checkpoint when reaching
```
Comment (tdomhan) [inline test/integration/test_seq_copy_int.py:54]: @fhieber  Did you have something alone these lines in mind?

Comment (fhieber) [reply test/integration/test_seq_copy_int.py:54]: yep, but typo: 'se' -> 'set'

Comment (tdomhan) [reply test/integration/test_seq_copy_int.py]: fixed
